include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -471,6 +471,81 @@ def NVMMASharedEncodingAttr :
   let hasCustomAssemblyFormat = 1;
 }
 
+def SwizzledBlocksSharedEncodingAttr :
+  TritonGPU_Attr<"SwizzledBlocksSharedEncoding", "swizzled_blocks_shared_encoding", [SharedEncodingTrait, LayoutEncodingTrait]> {
+  let mnemonic = "swizzled_blocks_shared";
+
+  let description = [{
+This shared encoding is similar to SwizzledSharedEncodingAttr,
+but instead of repeating swizzling pattern every `blockSize = maxPhase*perPhase` rows of memory object,
+this layout changes swizzling pattern.
+
+Let's consider an element with position = (row, col).
+For simplicity, we do not vectorize memory,
+i.e. vec == 1 and layout swizzles inidividual elements.
+
+Swizzled coordinates within memory object are (row, swizzledCol):
+
+  phase   = (row / perPhase) % maxPhase
+  blockNo = (row / perPhase) / maxPhase % maxPhase
+  combinedPhase = phase ^ blockNo
+  swizzledCol   = col ^ combinedPhase
+
+Swizzling examples (matrix is filled with numbers 0, 1, 2, .. col*row-1):
+
+  #shared<{vec=1, perPhase=1, maxPhase=2}>
```

Comment:
What are the `order` of these examples?

Reply:
these diagrams arrange fastest dimension in rows. I intentionally removed order to simplify description, seems this only brings more questions.

Let's consider this order = [1, 0], and I will add a comment about mapping of dimensions in these examples.